lib/web_ui/lib/painting.dart

```diff
@@ -491,6 +491,50 @@ Future<Codec> instantiateImageCodecFromBuffer(
   targetHeight: targetHeight,
   allowUpscaling: allowUpscaling);
 
+class _SizeOnlyImageDescriptor implements ImageDescriptor {
+  const _SizeOnlyImageDescriptor(this.width, this.height);
+
+  @override
+  final int width;
+
+  @override
+  final int height;
+
+  @override
+  dynamic noSuchMethod(Invocation invocation) => throw UnsupportedError('ImageDescriptor.${invocation.memberName} is not supported on web within a TargetImageSizeCallback.');
```

Comment:
Nooooooooo, can we avoid using noSuchMethod? :)

Reply:
TIL!  I'll update it.  Reason I went with this is because it was forcing me to implement all the private methods too, which was a bummer.

Reply:
Done - I got rid of the need for the ImageDescriptor at all.